src/eva/data/csv_space.py

```diff
@@ -0,0 +1,153 @@
+from eva.data.eva_dataset_base import EvaDatasetBase
+from eva.utilities.config import get
+from datetime import datetime
+import xarray as xr
+import numpy as np
+import csv
+
+
+class CsvSpace(EvaDatasetBase):
+
+    """
+    A class for handling CSV dataset configuration and processing.
+    """
+
+    def execute(self, dataset_config, data_collections, timing):
+
+        """
+        Executes the processing of CSV dataset.
+
+        Args:
+            dataset_config (dict): Configuration dictionary for the dataset.
+            data_collections (DataCollections): Object for managing data collections.
+            timing (Timing): Timing object for tracking execution time.
+        """
+
+        # Set the collection name
+        # -----------------------
+        collection_name = get(dataset_config, self.logger, 'name')
+
+        # Filename to be read into this collection
+        filenames = get(dataset_config, self.logger, 'filenames')
+
+        # get 'groups'
+        groups = get(dataset_config, self.logger, 'groups')
+
+        # Set the collection name
+        collection_name = dataset_config['name']
+
+        ds_list = []
+
+        # Read in the CSV files
+        for file in filenames:
+
+            f = open(file, "r")
+            fd = list(csv.reader(f, delimiter=","))
+            file_data = [[element.strip() for element in row] for row in fd]
```

Comment:
Can we put in an option that handles if there is a header or not?

Reply:
Yes that's a good idea.  The yaml file will have to specify the header.  I will add that.